contracts/cw4-group/src/helpers.rs

```diff
@@ -0,0 +1,62 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+use cosmwasm_std::{to_binary, Api, CanonicalAddr, CosmosMsg, HumanAddr, StdResult, WasmMsg};
+use cw4::{Cw4Contract, Member};
+
+use crate::msg::HandleMsg;
+
+/// Cw4GroupContract is a wrapper around HumanAddr that provides a lot of helpers
+/// for working with cw4-group contracts.
+///
+/// It extends Cw4Contract to add the extra calls from cw4-group.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Cw4GroupContract(pub Cw4Contract);
```

Comment:
Is this nesting really needed? / useful?

Reply:
There is a `Deref` so we "inherit" all the query functionality of `Cw4Contract`. That saves us from re-implementing that logic every step.

Reply:
> There is a `Deref` so we "inherit" all the query functionality of `Cw4Contract`. That saves us from re-implementing that logic every step.

Nice, but the logic is not much. Maybe it can be implemented below directly.

Reply:
I like this "inheritance" property of Tuple structs, by the way.

If I'm not mistaken, you still have to wrap the methods in order to expose them. It would be better to expose the public methods of the nested struct directly. And allow for overloading. But, at least traits are inherited, which is better than in the case of plain composition. Slowly moving to real inheritance... :-)

Reply:
> There is a `Deref` so we "inherit" all the query functionality of `Cw4Contract`. That saves us from re-implementing that logic every step.

I think I misunderstood you before. What you're saying is that, precisely, with that `Deref` trick we don't need to implement wrappers anymore. Cool, I'll try this. Thanks.

Reply:
Works like a charm. This is an "unofficial" "anti" pattern called [Deref polymorphism](https://github.com/rust-unofficial/patterns/blob/master/anti_patterns/deref.md), by the way.

I wish all unofficial things were as useful and convenient as this. Please, Rust people, more of these "unofficial" stuff, please.

Reply:
Poor man's inheritance, I'll call it.